bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java

```diff
@@ -139,6 +140,19 @@ public void receivedMessage(ThingUID connectionBridge, MqttBrokerConnection conn
             return;
         }
 
+        BaseChannelConfiguration config = BaseChannelConfiguration
+                .fromString(new String(payload, StandardCharsets.UTF_8), gson);
+
+        synchronized (configHashPerTopic) {
+            if (!configHashPerTopic.containsKey(topic)) {
+                configHashPerTopic.put(topic, config.hashCode());
+            } else {
+                if (configHashPerTopic.get(topic).equals(config.hashCode())) {
+                    return;
+                }
+            }
+        }
```

Comment:
This is simpler and only hits the map once. This change in combination with changing `configHashPerTopic` to a ConcurrentHashMap will make this just as thread-safe as using `synchronized`.
```suggestion
        Integer newHash = config.hashCode();
        Integer oldHash = configHashPerTopic.putIfAbsent(newHash);
        if(newHash.equals(oldHash)){
            return;
        }
```

Reply:
Thanks, I learn a lot from your comments!